### PR TITLE
[ISSUE #10] Add ding talk sink

### PR DESCRIPTION
**Pull request by zhaohai1299002788**
add ding talk sink

### CODE REVIEW

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/resources/log4j.properties [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+log4j.appender.console.target=System.err
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
+
+log4j.logger.org.apache.spark.repl.Main=WARN
+
+# Settings to quiet third party logs that are too verbose
+log4j.logger.org.sparkproject.jetty=WARN
+log4j.logger.org.sparkproject.jetty.util.component.AbstractLifeCycle=ERROR
+log4j.logger.org.apache.spark.repl.SparkIMain$exprTyper=INFO
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/resources/log4j.properties]: delete logback

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    }
+
+    @Override
+    public List<KeyValue> taskConfigs(int maxTasks) {
+        List<KeyValue> taskConfigList = new ArrayList<>(11);
+        KeyValue keyValue = new DefaultKeyValue();
+        keyValue.put(DingTalkConstant.WEB_HOOK, webHook);
+        keyValue.put(DingTalkConstant.MSG_TYPE_CONSTANT, msgType);
+        taskConfigList.add(keyValue);
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java:35]: delete duplicate code

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkTask.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public void resume() {
+
+    }
+
+    @Override
+    public void validate(KeyValue config) {
+        if (StringUtils.isBlank(config.getString(DingTalkConstant.WEB_HOOK))) {
+            throw new RuntimeException("ding talk required parameter is null !");
+        }
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkTask.java]: Added check webhook.

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkTask.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    private String webHook;
+
+    private String msgType;
+
+    @Override
+    public void put(List<ConnectRecord> sinkRecords) throws ConnectException {
+        try {
+            sinkRecords.forEach(sinkRecord -> {
+                Map<String, Object> objectMap = new HashMap<>();
+                objectMap.put(DingTalkConstant.CONTENT_CONSTANT, sinkRecord.getData());
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/com/aliyun/rocketmq/connect/dingtalk/sink/DingTalkSinkTask.java]: Added trasform configuration.

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/common/OkHttpUtils.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                paramMap.forEach(formBody::add);
+            }
+            requestBody = formBody.build();
+        }
+        request = new Request.Builder().post(requestBody).url(url);
+        return this;
+    }
+
+    public OkHttpUtils postForStringBody(Object data) {
+        String json = JSON.toJSONString(data);
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/common/OkHttpUtils.java:146]: 这里只需要原样将数据传递即可，不需要再JSON序列化：
 RequestBody requestBody = RequestBody.create(MediaType.parse("application/json; charset=utf-8"), data);

Comment (2011shenlin) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/common/OkHttpUtils.java:146]: Object使用它默认的ToString 方法

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        return taskConfigList;
+    }
+
+    @Override
+    public Class<? extends Task> taskClass() {
+        return DingTalkSinkTask.class;
+    }
+
+    @Override
+    public void validate(KeyValue config) {
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java:47]: 这个方法加个单测

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/common/OkHttpUtils.java (hunk 5 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/common/OkHttpUtils.java]: 已修改

File: connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java (hunk 6 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-dingtalk/src/main/java/org/apache/rocketmq/connect/dingtalk/sink/DingTalkSinkConnector.java:47]: 已添加

**Pull request merged**